src/conventions/h-group/clue-interpretation/connecting-cards.js

```diff
@@ -192,62 +192,39 @@ function find_unknown_connecting(game, giver, target, reacting, identity, looksD
  * @param {import('../../../types.js').ClueAction} clue
  * @returns {Connection[]}
  */
-export function resolve_bluff(game, connections, focusedCard, focusIdentity, clue) {
+export function resolve_bluff(game, target, connections, focusedCard, focusIdentity, clue) {
 	if (connections.length == 0 || !connections[0].bluff)
 		return connections;
 
-	const promised = connections.filter(conn => !conn.hidden).map(conn => conn.card).concat([focusedCard]);
-	const { state } = game;
+	const firstPlay = connections.findIndex(conn => !conn.hidden);
 	const bluffCard = connections[0].card;
 	let bluff_fail_reason = undefined;
 
-	// Determine the next play if this is a bluff.
-	const next_play = connections.findIndex(connection => connection.card.order == promised[0].order) + 1;
-
 	// Disallow self-colour bluffs: https://hanabi.github.io/extras/special-bluffs#self-color-bluffs-1-for-1-form-scb
-	if ((clue.giver + 1) % game.state.numPlayers == clue.target && clue.clue.type == CLUE.COLOUR && state.hands[clue.target].find(c => c.order == connections[0].card.order))
+	if ((clue.giver + 1) % game.state.numPlayers == clue.target && clue.clue.type == CLUE.COLOUR && connections[0].reacting == clue.target)
 		bluff_fail_reason = `self-colour bluff disallowed at level ${game.level}`;
-
+	
 	// A bluff must be followed only by prompts as otherwise it would not have been a valid bluff target.
-	if (!bluff_fail_reason && connections.some((conn, index) => index >= next_play && (conn.hidden || conn.type === 'finesse')))
+	if (!bluff_fail_reason && connections.some((conn, index) => index > firstPlay && (conn.hidden || conn.type === 'finesse')))
 		bluff_fail_reason = `requires additional finesses [${connections.map(logConnection)}]`;
 
-	if (!bluff_fail_reason) {
-		// A bluff must be recognizable. As such, there should be no connection
-		// between the bluffed card and at least one of the following plays
-		// as known by the player who would play next after the bluff play.
-		const next_player = state.hands.findIndex(hand => hand.findOrder(promised[1].order));
-		const bluff_identities = bluffCard.identity() === undefined ?
-			game.players[state.ourPlayerIndex].thoughts[bluffCard.order].inferred.filter(c => state.isPlayable(c)) :
-			[bluffCard];
-
-		const known_bluff = promised.some((card, i) =>
-			i > 0 && bluff_identities.some(({ suitIndex, rank }) => {
-				const expected = { suitIndex , rank: rank + i };
-
-				if (state.hands[next_player].findOrder(card.order))
-					return !game.players[next_player].thoughts[card.order].inferred.has(expected);
-				else
-					return !card.matches(expected);
-			}));
-
-		if (!known_bluff)
-			bluff_fail_reason = `${game.state.playerNames[next_player]} would not recognize the bluff`;
-	}
+	if (!bluff_fail_reason && bluffCard.identity() !== undefined && game.players[target].thoughts[focusedCard.order].inferred.has({suitIndex: bluffCard.suitIndex, rank: bluffCard.rank + 1}))
```

Comment:
I'm not sure if simply checking the inferences is sufficient, you might need to check the clue. For example, in a game with a pink suit, cluing 3 and having an i1 blind play should not connect, even though the 3-clued card may have i2 as an inference.

Reply:
Done, added a test covering this exact case which now passes.